spec/index.md

```diff
@@ -5033,6 +5101,215 @@ S with
 
 ```
 
+#### IC Management Canister: Take canister snapshot
+
+Only the controllers of the given canister can take a snapshot.
+
+```html
+
+S.messages = Older_messages · CallMessage M · Younger_messages
+(M.queue = Unordered) or (∀ msg ∈ Older_messages. msg.queue ≠ M.queue)
+M.callee = ic_principal
+M.method_name = 'take_canister_snapshot'
+M.arg = candid(A)
+M.caller ∈ S.controllers[A.canister_id]
+
```

Comment:
We should check here that no snapshot is stored for a canister or A.replace_snapshot is set to the snapshot ID of the currently stored snapshot.